### PR TITLE
[SYNCOPE-1609] Changes jsonb queries in order to improve performances

### PR DESCRIPTION
**Pull request by fmartelli**
Improve queries in order to avoid table joins

### CODE REVIEW

File: core/persistence-jpa-json/src/main/resources/pgjsonb/views.xml [hunk 1]
```diff
@@ -47,13 +47,13 @@ under the License.
   </entry>
 
   <!-- user -->
-  <entry key="user_search">
+<!--  <entry key="user_search">
```
Comment (ilgrosso) [inline core/persistence-jpa-json/src/main/resources/pgjsonb/views.xml:50]: Why comment this entry? If not needed anymore, please just remove.

File: core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java [hunk 2]
```diff
@@ -18,76 +18,57 @@
  */
 package org.apache.syncope.core.persistence.jpa.dao;
 
+import static org.apache.syncope.core.persistence.jpa.dao.AbstractAnySearchDAO.RELATIONSHIP_FIELDS;
```
Comment (ilgrosso) [inline core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java:21]: These two static imports shall be removed

File: core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java [hunk 3]
```diff
@@ -18,76 +18,57 @@
...
-                }
-
-                where.append(')');
-            } else {
-                where.append(',').append(searchView.name);
-            }
-            where.append(' ').append(searchView.alias);
-        });
-    }
+    protected static final String EMPTY_QUERY = "1=2";
```
Comment (ilgrosso) [inline core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java:71]: There is already a constant defined in superclass (`JPAAnySearchDAO`) with same name, please rename.

File: core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java [hunk 4]
```diff
@@ -18,76 +18,57 @@
...
-import org.apache.syncope.core.provisioning.api.serialization.POJOHelper;
-import org.apache.syncope.core.persistence.api.entity.JSONPlainAttr;
+import org.apache.syncope.core.persistence.api.entity.Realm;
 import org.apache.syncope.core.provisioning.api.utils.FormatUtils;
+import org.apache.syncope.core.provisioning.api.utils.RealmUtils;
 
 public class PGJPAJSONAnySearchDAO extends AbstractJPAJSONAnySearchDAO {
 
-    @Override
-    protected void processOBS(
```
Comment (ilgrosso) [inline core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java:43]: What about this method? Is it going to be removed completely? There might be subclasses with override: what should be done in such cases?

**Review by ilgrosso — CHANGES_REQUESTED** [review]

**Comment by ilgrosso** [comment]
@fmartelli would it be possible to:

1. open an improvement on JIRA describing the changes you want to apply
2. restore ordering of methods in `PGJPAJSONAnySearchDAO` so that it's easier to compare with current version?

Thanks!

File: core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java (hunk 4 above)
Comment (fmartelli) [reply core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java:43]: I didn't understand the question. 
PGJPAJSONAnySearchDAO does not need processOBS anymore while other classes need it.
Could you articulate this point?

**Comment by fmartelli** [comment]
I've just opened the improvement 
https://issues.apache.org/jira/browse/SYNCOPE-1609

File: core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java (hunk 4 above)
Comment (ilgrosso) [reply core/persistence-jpa-json/src/main/java/org/apache/syncope/core/persistence/jpa/dao/PGJPAJSONAnySearchDAO.java:43]: Nope, it is fine now, thanks!

**Review by ilgrosso — APPROVED** [review]
Great work, thanks! :+1:

**Pull request merged**